mesonbuild/interpreter/type_checking.py

```diff
@@ -563,7 +563,8 @@ def _objects_validator(vals: T.List[ObjectTypes]) -> T.Optional[str]:
             non_objects.extend(o for o in val.get_outputs() if not compilers.is_object(o))
 
     if non_objects:
-        return f'File{"s" if len(non_objects) > 1 else ""}: "{", ".join(non_objects)}" are not objects'
+        s = "s" if len(non_objects) > 1 else ""
+        return f'File{s}: "{", ".join(non_objects)}" are not objects'
```

Comment:
TBH I don't think being grammatically correct in an error message is world's first problem... but if you want to conditionally add an "s" to File, you also have to change are/is and make singular "object"... I would personally make that message unconditionally singular or plural as you wish.

Another nitpicking, I find it more readable as:
```python
non_objects_str = ', '.join(non_objects)
return f'Files {non_objects_str!r} are not objects'
```

Reply:
The more I think about it the more I don't like even having `Files`. Let me know what you think now, there's so little left I'd prefer to leave the join in the f-string, but I guess I can be convinced